paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -934,27 +935,52 @@ def autoscale_local_cluster(config_folder, dry_run=False, log_level=None):
     autoscaling_resources = system_config.get_cluster_autoscaling_resources()
     autoscaling_draining_enabled = system_config.get_cluster_autoscaling_draining_enabled()
     all_pool_settings = system_config.get_resource_pool_settings()
-    autoscaling_scalers = []
     mesos_state = get_mesos_master().state
     utilization_errors = get_all_utilization_errors(autoscaling_resources, all_pool_settings, mesos_state)
+    autoscaling_scalers = defaultdict(list)
+    any_cancelled_pool = defaultdict(lambda: False)
+    any_scale_up_pool = defaultdict(lambda: False)
     for identifier, resource in autoscaling_resources.items():
         pool_settings = all_pool_settings.get(resource['pool'], {})
         try:
-            autoscaling_scalers.append(get_scaler(resource['type'])(
+            scaler = get_scaler(resource['type'])(
                 resource=resource,
                 pool_settings=pool_settings,
                 config_folder=config_folder,
                 dry_run=dry_run,
                 log_level=log_level,
                 utilization_error=utilization_errors[(resource['region'], resource['pool'])],
                 draining_enabled=autoscaling_draining_enabled,
-            ))
+            )
+            autoscaling_scalers[resource['pool']].append(scaler)
+            any_scale_up_pool[resource['pool']] |= utilization_errors[(resource['region'], resource['pool'])] > 0
```

Comment:
This line is a bit silly now that I think of it...we can figure out this info just by looking at the utilization error for the pool.  So the first part of this pull request is redundant, since we won't try to do anything with cancelled resources anyways if the pool needs to scale up.